src/extensions.ts

```diff
@@ -10,3 +10,7 @@ export const EP_PHOVEA_CORE_LOGIN = 'epPhoveaCoreLogin';
  */
 export const EP_PHOVEA_CORE_LOGOUT = 'epPhoveaCoreLogout';
 
+/**
+ * Load locale JSON files with translations
+ */
+export const EP_PHOVEA_CORE_LOCALE = 'epPhoveaCoreLocale';
```

Comment:
Add interface and documentation:

```ts
export interface ILocaleEPDesc extends IPluginDesc {
  /**
   * Locale namespace that usually matches the resource filename (e.g., `phovea` and `phovea.json`)
   */
  ns: string;

  /**
   * Language following the [IETF language tag](https://en.wikipedia.org/wiki/IETF_language_tag) (e.g., `en` or `en-US`)
   */
  lng: string;

  /**
   * Order of the locales, if two locales with the same namespace and language are found.
   * Locales with a higher the order number override the locales with lower order numbers.
   */
  order: number;
}
```

Reply:
@oltionchampari Should we make `order` optional (i.e., defaults to `0`)? Should we further make `lng` optional as it would default to `en`? And then should we make `ns` optional?

Reply:
- I made order, lng, ns optional, and assigned default values if not  specified .
- If no ns is specified it gets added to the default namespace.
- In that case you can use the syntax `i18n.t('key')` instead of `i18n.t('namespace:key')`.